internal/auth/auth.go

```diff
@@ -62,19 +65,42 @@ func NewAuthService(logger logr.Logger, cluster *kubernetes.Cluster) *AuthServic
 	}
 }
 
-// GetUsers returns all the Epinio users
-func (s *AuthService) GetUsers(ctx context.Context) ([]User, error) {
+// GetUsers returns all the Epinio users with no conflicting definitions. it further returns a map
+// of definition counts enabling the caller to distinguish between `truly does not exist` versus
+// `has conflicting definitions`.
+func (s *AuthService) GetUsers(ctx context.Context) ([]User, DefinitionCount, error) {
 	s.Logger.V(1).Info("GetUsers")
 
 	secrets, err := s.getUsersSecrets(ctx)
 	if err != nil {
-		return nil, errors.Wrap(err, "error getting users secrets")
+		return nil, nil, errors.Wrap(err, "error getting users secrets")
 	}
 
 	users := []User{}
 	usernames := []string{}
 
+	// map of per-user definition counts, to detect users with conflicting definitions
+	userCount := DefinitionCount{}
+
+	for _, secret := range secrets {
```

Comment:
I'm commenting here but it's about the two loops. It's probably easier to read if we fetch all the users, mapping the duplicates, and then filter them.

Please be aware that the `newUserFromSecret` is suppose to hide all the "secret-user" internal, to avoid knowing from other parts which data contains what (ie: `secret.Data["username"]`). This is helpful if we change something in the secrets, and to avoid typos.

```go
allUsers :=  []User{}
usernames := []string{}
userCount := DefinitionCount{}

// fetch and convert all the users, mapping the duplicates
for _, secret := range secrets {
	user := newUserFromSecret(s.Logger, secret)
	allUsers = append(allUsers, user)
	usernames = append(usernames, user.Username)

	count := userCount[user.Username]
	userCount[user.Username] = count + 1
}

filteredUsers :=  []User{}

// filter duplicates
for _, user := range allUsers {
	if userCount[user.Username] > 1 {
		s.Logger.V(1).Info("skip duplicate user", "user", user.Username)
		continue
	}
	filteredUsers = append(filteredUsers, user)
}
```

Reply:
I had hoped to avoid the construction of `User` structures for the conflicting parts. That said, with the encapsulation argument, it is better to do so. Ok

Reply:
done. commit b04ac0d8192ab565f309eb40f53dc5acf34710cb